psutil/_psutil_osx.c

```diff
@@ -1798,6 +1799,278 @@ psutil_cpu_stats(PyObject *self, PyObject *args) {
 }
 
 
+/*
+ * Return temperatures of cpu cores
+ */
+static PyObject *
+cpu_cores_temperatures(PyObject *self, PyObject *args) {
+	smc_key smc_keys[] = {
+		{"TC%dC\0", "Core %d"},
+	};
+    int physicalcpus;
+    size_t size = sizeof(int);
+    int i;
+    float core_temp;
+    char key[16];
+    PyObject *py_tuple = NULL;
+    PyObject *py_retlist = PyList_New(0);
+
+    if (py_retlist == NULL)
+        return NULL;
+    if (sysctlbyname("hw.physicalcpu", &physicalcpus, &size, NULL, -1))
+        Py_RETURN_NONE;  // mimic os.cpu_count()
+    for (i = 0; i < physicalcpus + 1; i++) {
+		// +1 as it appears on some systems, core count starts at 1
+        sprintf(key, smc_keys[0].key, i);
+        core_temp = SMCGetTemperature(key);
+		// Problem if CPU actaully reaches negative temperature
+        if ((int)core_temp <= 0) {
+            continue;
+        }
+
+        sprintf(key, "Core%d", i);
+        py_tuple = Py_BuildValue(
+            "(sf)",
+            key,               // lable
+            core_temp          // value
+        );
+        if (!py_tuple)
+            goto error;
+
+        if (PyList_Append(py_retlist, py_tuple)) {
+            goto error;
+        }
+        Py_XDECREF(py_tuple);
```

Comment:
Suggestion: these lines (from `py_tuple` to here) are recurring and can be put in a utility function like `int _append_pair(PyObject *py_retlist, char key, char value)`